FtpClientApp/FtpClientApp/FileUpload.cs

```diff
@@ -1,10 +1,83 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Text;
+using System.IO;
+using System.Linq;
+using System.Net;
 
 namespace FtpClientApp
 {
-    class FileUpload
+    //Class for uploading a file on the remote FTP server 
+    //Works off of the functions provided in Program.cs
+    public class FileUpload
     {
+        //A MainMenu variable to keep track of the user and server for the use of an instance of this class.
+        private ServerConnectionInformation connection;
+        //private FtpWebRequest request;
+        private WebClient request;
+        private FtpTestWrapper wrapper;
+
+        //A constructor for the class which takes in a ServerconnectionInformation to set up for its use.
+        public FileUpload(ServerConnectionInformation toUse)
+        {
+            this.connection = toUse;
+
+        }
+
+        //A function to get File name to be uploaded & location where file is to eb stored. 
+        public String getFileName()
+        {
+            String filepath = Console.ReadLine();
+            return filepath;
+        }
+
+        //setup() function takes 2 strings as input: file to be uploaded and location on server 
+        //returns a String which will be 'success' in the case File is successfully uploaded 
+        //Otherwise the string will contain a relevant error message.
+        public String setup(String filetobeuploaded, String locationonserver)
+        {
+            try
+            {
+                if (!File.Exists(filetobeuploaded))
+                {
+                    Console.WriteLine("File does not exist. Please enter valid file path");
+                    return "disconnect";
+                }
+                String input = filetobeuploaded;
+                String pattern = @"/";
+                String[] elements = System.Text.RegularExpressions.Regex.Split(input, pattern);
+                string lastItem = elements[elements.Length - 1];
+                Console.WriteLine(lastItem);
+
+                String serverpath = locationonserver + "/" + lastItem;
+
+                String extension = Path.GetExtension(lastItem);
+                //Console.WriteLine("GetExtension('{0}') returns '{1}'", lastItem, extension);
+                if ((extension != ".txt") && (extension != ".jpg") && (extension != ".png"))
+                {
+                    Console.WriteLine("Please enter one of the following file formats only :.txt, .jpg, .png");
+                    return "disconnect";
```

Comment:
It would be better just to return a boolean value instead of returning the strings "success" and "failure." Returning a string would be helpful in the case there are various return options (maybe 5+) but just to return if the function succeeded or failed should be a boolean.

Reply:
Take a look at my ListDirectoryLocal class file if you would like an example of what I mean. Look how I use the local variable `bool success=false;`

Reply:
I checked that, but  i wanted to keep the coding style consistent with how all other functions are (CreateRemotedir,, ListFiles, FileDownload etc.) which all have return "success" or "disconnect", so i kept it as is.